src/sagemaker_xgboost_container/algorithm_mode/handler_service.py

```diff
@@ -0,0 +1,197 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+import numpy as np
+import os
+import pickle as pkl
+from scipy.sparse import csr_matrix
+import xgboost as xgb
+
+from sagemaker_inference import content_types, default_inference_handler, encoder
+from sagemaker_inference.default_handler_service import DefaultHandlerService
+
+from sagemaker_xgboost_container.algorithm_mode.mms_transformer import Transformer
+from sagemaker_xgboost_container import encoder as xgb_encoder
+from sagemaker_xgboost_container.algorithm_mode.inference_errors import NoContentInferenceError, \
+    UnsupportedMediaTypeInferenceError, ModelLoadInferenceError, BadRequestInferenceError
+
+
+SAGEMAKER_BATCH = os.getenv("SAGEMAKER_BATCH")
+
+
+# FIXME: https://github.com/aws/sagemaker-xgboost-container/issues/12
+# This was failing a previous container test; need to make decision as to change test or behavior.
+def _get_sparse_matrix_from_libsvm(payload):
+    pylist = map(lambda x: x.split(' '), payload.split('\n'))
+    colon = ':'
+    row = []
+    col = []
+    data = []
+    for row_idx, line in enumerate(pylist):
+        for item in line:
+            if colon in item:
+                col_idx = item.split(colon)[0]
+                val = item.split(colon)[1]
+                row.append(row_idx)
+                col.append(col_idx)
+                data.append(val)
+
+    row = np.array(row)
```

Comment:
don't we need a `.astype(np.int)` here?

Reply:
Seems to pass previous container test behavior